src/System.Runtime.Serialization.Xml/tests/Canonicalization/CanonicalizationTestHelper.cs

```diff
@@ -0,0 +1,189 @@
+        {
+            return CreateXmlReader(rwType, stream, encoding, quotas, dictionary, null);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, byte[] buffer, Encoding encoding, XmlDictionaryReaderQuotas quotas)
+        {
+            return CreateXmlReader(rwType, buffer, encoding, quotas, null);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, Stream stream, Encoding encoding, XmlDictionaryReaderQuotas quotas)
+        {
+            return CreateXmlReader(rwType, stream, encoding, quotas, null);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, byte[] buffer, Encoding encoding)
+        {
+            return CreateXmlReader(rwType, buffer, encoding, XmlDictionaryReaderQuotas.Max);
+        }
+
+        public static XmlReader CreateXmlReader(ReaderWriterType rwType, Stream stream, Encoding encoding)
+        {
+            return CreateXmlReader(rwType, stream, encoding, XmlDictionaryReaderQuotas.Max);
+        }
+
+        public static XmlWriter CreateXmlWriter(ReaderWriterType rwType, Stream stream, Encoding encoding)
+        {
+            return CreateXmlWriter(rwType, stream, encoding, null);
+        }
+
+        public static XmlWriter CreateXmlWriter(ReaderWriterType rwType, Stream stream, Encoding encoding, IXmlDictionary dictionary)
+        {
+            XmlWriter result = null;
+            switch (rwType)
+            {
+                case ReaderWriterType.Binary:
+                    result = XmlDictionaryWriter.CreateBinaryWriter(stream, dictionary);
+                    break;
+                case ReaderWriterType.Text:
+                    result = XmlDictionaryWriter.CreateTextWriter(stream, encoding);
+                    break;
+                case ReaderWriterType.WebData:
+                    XmlWriterSettings settings = new XmlWriterSettings();
+                    settings.Encoding = encoding;
+                    result = XmlWriter.Create(stream, settings);
+                    break;
+                case ReaderWriterType.WrappedWebData:
+                    XmlWriterSettings settings2 = new XmlWriterSettings();
+                    settings2.Encoding = encoding;
+                    result = XmlWriter.Create(stream, settings2);
+                    result = XmlDictionaryWriter.CreateDictionaryWriter(result);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(rwType));
+            }
+            return result;
+        }
+    }
+
+    public class Helper
+    {
+        public static void DumpToFile(string fileName, byte[] buffer, int offset, int count)
+        {
+            try
+            {
+                FileStream fs = File.Create(Path.Combine(Path.GetTempPath(), fileName));
```

Comment:
This is going to leave behind the temp file after the test completes. Can you please take a look at https://github.com/dotnet/corefx/pull/19142? We can do something similar as the PR does.